extensions/amp-access/0.1/amp-access.js

```diff
@@ -214,6 +217,10 @@ export class AccessService {
       user.warn(TAG, 'Experiment "amp-access-server" is not enabled.');
       type = AccessType.CLIENT;
     }
+    if (type == AccessType.CLIENT && this.isServerEnabled_) {
```

Comment:
You can do this, but then you need to make sure that `AmpAccessServer` forgoes encoding in the same conditions that it calls client adapter. E.g.

Current code:
```
  authorize() {
    ...
    if (!this.isProxyOrigin_ || !this.serverState_) {
      dev.fine(TAG, 'Proceed via client protocol');
      return this.clientAdapter_.authorize();
    }
  ...
```

Thus, we'll do also:
```
  encodeReaderId(readerId) {
    if (!this.isProxyOrigin_ || !this.serverState_) {
      dev.fine(TAG, 'Proceed via client protocol');
      return this.clientAdapter_.encodeReaderId(readerId);
    }
  ...
```

Basically, `authorize` and `encodeReaderId` should be always in sync. Generally, this is a good idea to do anyway. But it's especially problematic if we override `type=client` with `server` ourselves.